cfgov/v1/management/commands/update_data_snapshot_values.py

```diff
@@ -0,0 +1,79 @@
+import json
+import logging
+import os
+
+from django.core.management.base import BaseCommand
+
+from v1.models.browse_page import BrowsePage
+from v1.tests.wagtail_pages.helpers import publish_changes
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = 'Monthly updates to data snapshot values'
+
+    def expand_path(self, path):
+        """Expands a relative path into an absolute path"""
+        rootpath = os.path.abspath(os.path.expanduser(path))
+
+        return rootpath
+
+    def add_arguments(self, parser):
+        """Adds all arguments to be processed."""
+        parser.add_argument(
+            '--snapshot_file',
+            required=True,
+            help='JSON file containing all markets\' data snapshot values'
+        )
+
+    def get_data_snapshots(self):
+        """ Gets all data snapshots from browse pages
+        Assumes there is one data snapshot per page
+        """
+        snapshots = []
+        for page in BrowsePage.objects.all():
+            stream_data = page.specific.content.stream_data
+            snapshot = filter(
+                lambda item: item['type'] == 'data_snapshot',
+                stream_data
+            )
+            if snapshot:
+                snapshot[0]['value']['page'] = page
+                snapshots.append(snapshot[0]['value'])
+        return snapshots
+
+    def find_data_snapshot(self, market_key, snapshots):
+        """ Look up data snapshot by the provided market key
+        Assumes there is one data snapshot per key
+        """
+        snapshot = filter(
+            lambda item: item['market_key'] == market_key,
+            snapshots
+        )
+        if snapshot:
+            return snapshot[0]
+        return None
```

Comment:
That's a useful link, thanks! I see that Python recommends using list comprehensions instead of `filter` for cross compatibility.